packages/edit-site/src/components/dataviews/view-list.js

```diff
@@ -62,6 +66,41 @@ function HeaderMenu( { dataView, header } ) {
 	if ( ! isSortable && ! isHidable ) {
 		return text;
 	}
+	const isFilterable = header.column.columnDef.filters?.length > 0;
+	let filters = null;
+	if ( isFilterable ) {
+		filters = header.column.columnDef.filters.map( ( filter ) => {
```

Comment:
This and the UI components below is very similar to this code https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/dataviews/filters.js#L14

I wonder if the `Filters` component could render different children UI depending on the context (top-level vs columns, for example).